engine/src/main/java/org/hibernate/validator/internal/cfg/context/DefaultConstraintMapping.java

```diff
@@ -83,4 +91,31 @@ public final AnnotationProcessingOptionsImpl getAnnotationProcessingOptions() {
 
 		return configurations;
 	}
+
+	@Override
+	public <A extends Annotation> ConstraintDefinitionContext<A> constraint(Class<A> annotationClass) {
+		Contracts.assertNotNull( annotationClass, MESSAGES.annotationTypeMustNotBeNull() );
+		Contracts.assertTrue( annotationClass.isAnnotationPresent( Constraint.class ),
```

Comment:
Well, I don't see why we would check for null values and stop there. I guess there are null checks later in the pipeline too, but if we removed this null check here, the stacktrace would only be less clear.
That being said, indeed, I also stopped to what's simple to implement: I did not check the presence of the mandatory attributes on the annotation. I just wanted to make sure that the argument is what we expect (a constraint annotation), as opposed to validating the correctness of this annotation.

Thinking about it a little, if Java had allowed inheritance in annotations, there would probably have been a `ConstraintAnnotation` supertype, instead of a `@Constraint` annotation, so that you could have enforced the presence of the various annotation attributes. Thus I think the presence of the `@Constraint` annotation is almost some kind "type requirement", a contract between this API and its users.